docs/DevGuide/TensorExpressions.md

```diff
@@ -0,0 +1,411 @@
+\cond NEVER
+Distributed under the MIT License.
+See LICENSE.txt for details.
+\endcond
+# Writing tensor equations with TensorExpressions {#writing_tensorexpressions}
+
+\tableofcontents
+
+SpECTRE's `TensorExpression`s interface allows you to write tensor equations in
+SpECTRE in C++ with syntax that resembles tensor index notation. To use it,
+simply add this include to the top of your file:
+```
+#include "DataStructures/Tensor/Tensor.hpp"
+```
+The following guide assumes a basic understanding of the `Tensor` class and
```

Comment:
Yes, so I would love to. I tried many times to get it to link, but haven't been able to. I also looked elsewhere in spectre to find it successfully linked and I haven't found an instance yet, so I'm not sure how to get doxygen to link it. I think maybe the linking fails because the declaration of the class is not documented (not sure), and I can't get it to link to the defintiion, which is a partial template specialization and not sure how to tell doxygen what I want.

So if you happen to know how I could do this, that would be great, and then I think that needs to be done in a bunch of places throughout the site. Otherwise, I can keep trying things and poking around doxygen docs/issues

Reply:
e.g. I can try adding documentation tot he declaration of the `Tensor` class to see if that fixes it, unless you already know a solution

Reply:
I just tried moving the docs to the declaration and that fixed everything so it's all linked now. So I'd say just add a new commit before this one where you move the docs.

Reply:
Thanks! This is also useful to know for other missing links with the same problem and this should fix the linking to `Tensor` everywhere in spectre docs hopefullly